core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -130,20 +133,67 @@ class BillingProjectOrchestrator(ctx: RawlsRequestContext,
         )
       )
     } yield {}
+
+  def deleteBillingProjectV2(projectName: RawlsBillingProjectName): Future[Unit] =
+    for {
+      _ <- samDAO
+        .userHasAction(SamResourceTypeNames.billingProject,
+                       projectName.value,
+                       SamBillingProjectActions.deleteBillingProject,
+                       ctx
+        )
+        .flatMap {
+          case true => Future.successful()
+          case false =>
+            Future.failed(
+              new RawlsExceptionWithErrorReport(
+                errorReport = ErrorReport(StatusCodes.Forbidden, "You must be a project owner.")
+              )
+            )
+        }
+      billingProfileId <- billingRepository.getBillingProfileId(projectName)
+      projectLifecycle = billingProfileId match {
+        case None    => googleBillingProjectLifecycle
+        case Some(_) => bpmBillingProjectLifecycle
+      }
+      _ <- billingRepository.failUnlessHasNoWorkspaces(projectName)
+      _ <- projectLifecycle.preDeletionSteps(projectName, ctx)
+      _ <- unregisterBillingProjectWithUserInfo(projectName, ctx.userInfo)
+    } yield {}
+
+  // This code also lives in UserService.
+  def unregisterBillingProjectWithUserInfo(projectName: RawlsBillingProjectName,
+                                           ownerUserInfo: UserInfo
+  ): Future[Unit] =
+    for {
+      _ <- billingRepository.deleteBillingProject(projectName)
+      _ <- samDAO
+        .deleteResource(SamResourceTypeNames.billingProject,
+                        projectName.value,
+                        ctx.copy(userInfo = ownerUserInfo)
+        ) recoverWith { // Moving this to the end so that the rawls record is cleared even if there are issues clearing the Sam resource (theoretical workaround for https://broadworkbench.atlassian.net/browse/CA-1206)
```

Comment:
I don't understand the part about clearing the record - this doesn't appear to do anything aside from log the message and re-throw the exception

Reply:
This comment was already in the code I copied over… I think they are saying that they delete the Sam record after deleting the project so that the project is deleted even if the Sam call fails.